composeml/data_slice/offset.py

```diff
@@ -0,0 +1,128 @@
+import re
+
+import pandas as pd
+
+
+class DataSliceOffset:
+    """Offsets for calculating data slice indices."""
+    def __init__(self, value):
+        self.value = value
+        self._check()
+
+    def _check(self):
+        """Checks if the value is a valid offset."""
+        if isinstance(self.value, str): self._parse_value()
+        assert self._is_valid_offset, self._invalid_offset_error
+
+    @property
+    def _is_offset_base(self):
+        """Whether offset is a base type."""
+        return issubclass(type(self.value), pd.tseries.offsets.BaseOffset)
+
+    @property
+    def _is_offset_position(self):
+        """Whether offset is integer-location based."""
+        return pd.api.types.is_integer(self.value)
+
+    @property
+    def _is_offset_timedelta(self):
+        """Whether offset is a timedelta."""
+        return isinstance(self.value, pd.Timedelta)
+
+    @property
+    def _is_offset_timestamp(self):
+        """Whether offset is a timestamp."""
+        return isinstance(self.value, pd.Timestamp)
+
+    @property
+    def _is_offset_frequency(self):
+        """Whether offset is a base type or timedelta."""
+        value = self._is_offset_base
+        value |= self._is_offset_timedelta
+        return value
+
+    def __int__(self):
+        """Typecasts offset value to an integer."""
+        if self._is_offset_position: return self.value
+        elif self._is_offset_base: return self.value.n
+        elif self._is_offset_timedelta: return self.value.value
+        else: raise TypeError('offset must be position or frequency based')
+
+    def __float__(self):
+        """Typecasts offset value to a float."""
+        if self._is_offset_timestamp: return self.value.timestamp()
+        else: raise TypeError('offset must be a timestamp')
+
+    @property
+    def _is_positive(self):
+        """Typecasts offset value to an integer."""
```

Comment:
Docstring doesn't match output - needs to be updated.

Reply:
Inaccurate docstring was updated.